primitives/runtime-interface/proc-macro/src/runtime_interface/bare_function_interface.rs

```diff
@@ -45,17 +45,65 @@ use quote::{quote, quote_spanned};
 
 use std::iter;
 
+fn not_feature_force_version(
+	feature_force_version: &Vec<(String, String, u32)>,
+	method: &TraitItemMethod,
+) -> TokenStream {
+	let method = method.sig.ident.to_string();
+	for (feature, method_patch, _version_patch) in feature_force_version {
+		if &method == method_patch {
+			return quote! {
+				#[cfg(not(feature=#feature))]
+			}
+		}
+	}
+	quote! {}
+}
+
 /// Generate one bare function per trait method. The name of the bare function is equal to the name
 /// of the trait method.
-pub fn generate(trait_def: &ItemTrait, is_wasm_only: bool, tracing: bool) -> Result<TokenStream> {
+pub fn generate(
+	trait_def: &ItemTrait,
+	is_wasm_only: bool,
+	tracing: bool,
+	feature_force_version: &Vec<(String, String, u32)>,
+) -> Result<TokenStream> {
 	let trait_name = &trait_def.ident;
 	let runtime_interface = get_runtime_interface(trait_def)?;
 
 	// latest version dispatch
 	let token_stream: Result<TokenStream> = runtime_interface.latest_versions().try_fold(
 		TokenStream::new(),
 		|mut t, (latest_version, method)| {
-			t.extend(function_for_method(method, latest_version, is_wasm_only)?);
+			t.extend(function_for_method(
+				method,
+				latest_version,
+				is_wasm_only,
+				not_feature_force_version(&feature_force_version, &method),
+			)?);
+			Ok(t)
+		},
+	);
+
+	// forced version
+	let token_stream: Result<TokenStream> = feature_force_version.iter().try_fold(
+		token_stream?,
+		|mut t, (feature, method, force_version)| {
+			// lookup method
+			let (_, full_method) = runtime_interface
+				.all_versions()
+				.find(|(version, full_method)| {
+					version == force_version && &full_method.sig.ident.to_string() == method
+				})
+				.expect("Force version not found");
```

Comment:
This expect can be triggered by a user.
it is better to return a compile error than panic.

currently it would look like this:
```
error: custom attribute panicked
   --> primitives/io/src/lib.rs:101:1
    |
101 | #[runtime_interface(feature_force_version=old_state,root,3)]
    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
    = help: message: Force version not found
```